### PR TITLE
Assignment5

### PR DESCRIPTION
**Pull request by wpcwilliams**
Working on tests...

### CODE REVIEW

File: SecretSanta/src/SecretSanta.Domain/Services/PairingService.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            List<int> userIds = group?.GroupUsers?.Select(x => x.UserId).ToList();
+            if(userIds == null || userIds.Count < 2)
+            {
+                return null;
+            }
+
+            List<Pairing> pairings = await Task.Run( () => GetPairings(userIds));
+            if(pairings == null)
+                
+            await DbContext.Pairings.AddRangeAsync(pairings);
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Domain/Services/PairingService.cs]: There is a bug here. This statement is technically _inside_ of the `if(pairings == null)` above. So the pairings are never actually added to the database. Because the unit tests only check the returned pairings and not the values in the database this bug is not caught. There are also no unit tests that would cause `GetPairings` to return `null` which would have also caught this bug.

File: SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+                UserService uService = new UserService(context);
+                User user1 = new User { FirstName = "fname1", LastName = "lname1" };
+                User user2 = new User { FirstName = "fname2", LastName = "lname2" };
+                User user3 = new User { FirstName = "fname3", LastName = "lname3" };
+                User user4 = new User { FirstName = "fname4", LastName = "lname4" };
+                uService.AddUser(user1);
+                uService.AddUser(user2);
+                uService.AddUser(user3);
+                uService.AddUser(user4);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingServiceTests.cs]: These are async method calls and should have been awaited.

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/PairingControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        [TestMethod]
+        public async Task GeneratePairings_GroupDoesNotContainMembers()
+        {
+            //group doesn't contain members
+
+
+
+            Assert.AreEqual(0, 0);
+        }
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/PairingControllerTests.cs]: Would have been nice to see these tests get finished.

File: SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingController.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿namespace SecretSanta.Domain.Tests.Services
+{
+    internal class PairingController
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingController.cs]: Accidental class generation? You should not need controllers in the Domain level unit tests.

File: SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+
+                PairingService pService = new PairingService(context);
+
+                List<Pairing> pairings = await pService.GeneratePairings(group.Id);
+
+
+                foreach (Pairing p in pairings)
+                {
+                    Assert.AreNotEqual(p.Recipient, p.SantaId);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Domain.Tests/Services/PairingServiceTests.cs]: This is good, but it only checks one of the pairing requirements for being valid.
